### DISCUSSION TITLE
creating resources inflight

### DISCUSSION THREAD

**Discussion by vanjaoljaca** [winglang/wing#1090] [General]
What is the vision going forward for creating inflight resources? I'm really curious how the language and devxp will evolve in this regard. Is it considered something that will be in scope for the project, or will it be changing the design too much?

I think about this in relation to UnisonLang which also explores a similar area in a slightly different way.

**Comment by ekeren** [comment]
Hey @vanjaoljaca , 

Can you please point me to any UnisonLang links that describes this behavior?

**Comment by vanjaoljaca** [comment]
Unison lang/cloud video is here: https://www.unison.cloud/

The basic idea they run with is that all provisioned hosts in the cloud run the Unison runtime, and any code you write and run locally can be executed in the cloud with ```cloud.runAt remote myFunction```, the code is automatically sent to the remote host for running and the result is returned. I assume the vision there is that queues and 'dynamodb' style components will be deployed as a standard library using runAt. Just curious if winglang plans to keep the preflight/inflight hard separation forever, or if its a short term thing.
